### DISCUSSION TITLE
Video intensive platform powered by Supabase

### DISCUSSION THREAD

**Discussion by hld0** [supabase/supabase#26958] [Questions]
Hi!

I am currently working on a Flutter application for which the uploading and viewing of videos is a central feature. Admittedly, I do not have experience with a platform like this and I ran into some issues I did not even think of beforehand. I was wondering if anyone in the community had similar issues and how they resolved those.

My first problem is the egress. No matter how much you compress the videos you will run into problems as it will generate too much traffic. My solution to this is to download and store the videos locally in a temporary folder which seems to have worked in dramatically decreasing the necessary bandwidth.

The really big problem I have encountered though is compression. My initial assumption was that simply using on-device compression will do the trick but I ran into the following issues:

- Using `ffmpeg` might be a good choice due to its versatility, but it is incredibly slow and therefore not usable in this case (You can not expect users to wait a few minutes when uploading short 30 second videos).
- Using native compression libraries via flutter packages such as [video_compress](https://pub.dev/packages/video_compress) dramatically increases the compression speed but unfortunately allows only for very limited customization. On iOS one can choose between `LowQuality`, `MediumQuality` and `HighQuality`. The lowest setting reduces the video file size perfectly, but the quality is unusable unfortunately. The medium setting on the other hand simply produces files which are too large.

My reference for the video size I am aiming for are Instagram reels which seem to keep very acceptable video qualities with very low file sizes, e.g. an approximately 1 minute long video (when downloaded) has around 30 MB while an uncompressed iPhone video of the same length has around 500 MB.

While researching I found a project called [frostbyte](https://github.com/0xDebabrata/frostbyte) which offers transcoding services for Supabase. I haven't tested it yet though, as it seems quite raw. A feature request for a transcoding service #18002 has yet to receive an answer.

So to summarize I was wondering if anyone has used Supabase for a similar purpose and has maybe already resolved the same issues I am having.

Best wishes,
H

**Comment by tiltmaster** [comment]
I am wondering the same, and asking the same in Flutter Discord, Supabase discord, and zero responses or even details of this.
